src/main/java/org/example/dao/ProductDAO.java

```diff
@@ -0,0 +1,68 @@
+package org.example.dao;
+
+import org.example.model.Product;
+import org.example.model.User;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class ProductDAO extends AbstractDAO<Product> {
+    public ProductDAO(Connection connection) {
+        super(connection, "product");
+    }
+
+    public int getProductId(Product product) {
```

Comment:
This is not a comment, it's more a question. I'm wondering if it's  possible to get a product without an id field which is not defined in a Product.java. I just would like to learn if it's possible.

Reply:
Product Id is not defined in Product.java as it is auto generated when creating the object. I believe there is a way to execute a query to return the auto generated id as well with jdbc. I am not sure if that makes sense or answers your question, but basically, the database stores the id but the application doesn't necessarily need the id so I left it out.

Reply:
Oh yeah I got it. I'll try to research it. Thank you for your response.